processing/src/main/java/org/apache/druid/query/filter/AndDimFilter.java

```diff
@@ -72,8 +73,20 @@ public AndDimFilter(DimFilter... fields)
   @Override
   public DimFilter optimize()
   {
-    List<DimFilter> elements = DimFilters.optimize(fields);
-    return elements.size() == 1 ? elements.get(0) : new AndDimFilter(elements);
+    List<DimFilter> elements = DimFilters.optimize(fields)
+                                         .stream()
+                                         .filter(filter -> !(filter instanceof TrueDimFilter))
+                                         .collect(Collectors.toList());
+    if (elements.isEmpty()) {
+      // All elements were TrueDimFilter after optimization
+      return TrueDimFilter.instance();
+    } else if (elements.size() == 1) {
+      return elements.get(0);
```

Comment:
AndDimFilterTest is missing branch coverage for this and line 88

Reply:
Looks like this code is hit by other tests in druid-processing, but I'm not sure which ones

Reply:
The use of this and `OrDimFilter` are prevalent in unit tests so I believe they are being tested somewhere (I didn't change the logic of the lines you mentioned). But I can add some more tests.

Reply:
Added.